### ISSUE TITLE
Invalid thread access error caused by big-data-plugin when multiple sessions are present

### ISSUE THREAD

**Issue by HiromuHota** [HiromuHota/pentaho-kettle#23]
Here is the error message on the server side.
```
2017/01/11 13:55:44 - Spoon - ERROR (version 6.1.0.1-196, build 1 from 2016-04-07 12.08.49 by buildguy) : Error handling menu right click on job entry through extension point
2017/01/11 13:55:44 - Spoon - ERROR (version 6.1.0.1-196, build 1 from 2016-04-07 12.08.49 by buildguy) : org.eclipse.swt.SWTException: Invalid thread access
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.SWT.error(SWT.java:3720)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.SWT.error(SWT.java:3643)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.SWT.error(SWT.java:3614)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Widget.error(Widget.java:1018)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Widget.checkWidget(Widget.java:956)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Text.getText(Text.java:203)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.di.core.hadoop.HadoopClusterViewTreeExtension.filterMatch(HadoopClusterViewTreeExtension.java:144)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.di.core.hadoop.HadoopClusterViewTreeExtension.refreshNamedClusterSubtree(HadoopClusterViewTreeExtension.java:124)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.di.core.hadoop.HadoopClusterViewTreeExtension.callExtensionPoint(HadoopClusterViewTreeExtension.java:84)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.di.core.extension.ExtensionPointHandler.callExtensionPoint(ExtensionPointHandler.java:46)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.di.ui.spoon.Spoon.refreshSelectionTreeExtension(Spoon.java:6403)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.di.ui.spoon.Spoon.refreshTree(Spoon.java:6235)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.di.ui.spoon.Spoon$8.modifyText(Spoon.java:1972)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:202)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:109)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:687)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:594)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Text.setText(Text.java:184)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.di.ui.spoon.Spoon.setDesignMode(Spoon.java:2076)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.di.ui.spoon.Spoon.newJobFile(Spoon.java:4774)
2017/01/11 13:55:44 - Spoon - 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2017/01/11 13:55:44 - Spoon - 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2017/01/11 13:55:44 - Spoon - 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2017/01/11 13:55:44 - Spoon - 	at java.lang.reflect.Method.invoke(Method.java:498)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.ui.xul.impl.AbstractXulDomContainer.invoke(AbstractXulDomContainer.java:313)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.ui.xul.impl.AbstractXulComponent.invoke(AbstractXulComponent.java:157)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.ui.xul.impl.AbstractXulComponent.invoke(AbstractXulComponent.java:141)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.ui.xul.jface.tags.JfaceMenuitem.access$100(JfaceMenuitem.java:43)
2017/01/11 13:55:44 - Spoon - 	at org.pentaho.ui.xul.jface.tags.JfaceMenuitem$1.run(JfaceMenuitem.java:106)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.jface.action.Action.runWithEvent(Action.java:493)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:575)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.jface.action.ActionContributionItem.access$2(ActionContributionItem.java:492)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.jface.action.ActionContributionItem$5.handleEvent(ActionContributionItem.java:403)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:109)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:687)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:594)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Display.executeNextEvent(Display.java:1217)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Display.runPendingMessages(Display.java:1198)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Display.safeReadAndDispatch(Display.java:1181)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:1173)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.rap.rwt.application.AbstractEntryPoint.createUI(AbstractEntryPoint.java:69)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.rap.rwt.internal.lifecycle.RWTLifeCycle.createUI(RWTLifeCycle.java:177)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.rap.rwt.internal.lifecycle.RWTLifeCycle$UIThreadController.run(RWTLifeCycle.java:290)
2017/01/11 13:55:44 - Spoon - 	at java.lang.Thread.run(Thread.java:745)
2017/01/11 13:55:44 - Spoon - 	at org.eclipse.rap.rwt.internal.lifecycle.UIThread.run(UIThread.java:107)
```

**Comment by HiromuHota** [comment]
This happens only when at least one Hadoop cluster is configured in (web)Spoon.

**Comment by HiromuHota** [comment]
Resolved by HiromuHota/big-data-plugin@0298a4be601869fa5ab8ce7852d3a787953c78c1.
